File: Python/70-Climbing-Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [0 for i in range(n+1)]
        dp[0], dp[1] = 1, 1
        for i in range(2, n+1):
            dp[i] = dp[i-1]+dp[i-2]
        return dp[n]


s = Solution()
print(s.climbStairs(5))

File: JavaScript/70-Climbing-Stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n, dp = new Array()) {
  if (n == 1) return 1;
  if (n == 2) return 2;
  if (dp[n] != undefined) return dp[n];
  dp[n] = climbStairs(n - 1, dp) + climbStairs(n - 2, dp);
  return dp[n];
};
